### ISSUE TITLE
Citation IRIs as virtual resources

### ISSUE THREAD

**Issue by essepuntato** [essepuntato/opencitations#23]
The idea is to plan to provide a specific IRI for each citation included in the Corpus. However, we would like to avoid to add additional resources in the Corpus (i.e. a specific resource for each citation), since we want to avoid to be redundant with the data and to avoid the additional use of HD space.

Thus, the idea is to reserve a space in the OCC URLs for enabling "virtual" IRIs for accessing citations among two articles. Such "virtual" IRIs basically should not exist in the Corpus (nor in the dumps), but they could be accessed via HTTP as usual (by means of content negotiation). The basic structure of these virtual IRIs for citations could be as follows:

```
https://w3id.org/oc/virtual/citation/[citing OCC ID]/[cited OCC ID]
```

For instance, a citation from br/1 to br/11 would be identified by `https://w3id.org/oc/virtual/citation/br/1/br/11`.

If an agent asks for the information available to such citation virtual IRI, the serving mechanism would return all the basic string metadata (title, subtitle, year, if any) + type of the citing article, of the cited article, and the explicit link between those via `cito:cites`. For instance, considering the aforementioned example, accessing to `https://w3id.org/oc/virtual/citation/br/1/br/11` will result in having the following RDF triples (in Turtle here):

```
<https://w3id.org/oc/corpus/br/1> cito:cites <https://w3id.org/oc/corpus/br/11> .

<https://w3id.org/oc/corpus/br/1> a fabio:Expression , fabio:JournalArticle ;
    dcterms:title "ESMO-ESGO-ESTRO Consensus Conference on Endometrial Cancer" ;
    fabio:hasSubtitle "Diagnosis, Treatment and Follow-up" ;
    fabio:hasPublicationYear "2016"^^xsd:gYear .

<https://w3id.org/oc/corpus/br/11> a fabio:Expression , fabio:JournalArticle ;
    dcterms:title "ACOG Practice Bulletin #65: Management of Endometrial Cancer" ;
    ns5:hasPublicationYear "2005"^^xsd:gYear .
```

This is just a proposal still under discussion – please feel free to comment on it.

**Comment by essepuntato** [comment]
As alternative to

```https://w3id.org/oc/virtual/citation/[citing OCC ID]/[cited OCC ID]```

we could also used a more compact format, e.g.:

```https://w3id.org/oc/virtual/citation/[citing BR ID]-[cited BR ID]```

since citations in the OCC happen only between BR resources. For instance, a citation from br/1 to br/11 would be identified by `https://w3id.org/oc/virtual/citation/1-11`.

**Comment by davidshotton** [comment]
Additionally, using the proposed new CiTO class cito:Citation and data property cito:hasCitationSpan, the returned metadata could include statements defining the URI as a citation and defining the citation span, namely the interval (expressed in whole years) between the publication year of the cited entity and the publication year of the citing entity.

       <https://w3id.org/oc/virtual/citation/1-11> a cito:Citation ;
                                                              cito:hasCitationSpan 11 .

**Comment by davidshotton** [comment]
What does "ns5:hasPublicationYear "2005"^^xsd:gYear ." above mean? 
Shouldn't this be "fabio:hasPublicationYear "2005"^^xsd:gYear ."?

**Comment by essepuntato** [comment]
@davidshotton,

Yes, wrong cut&paste: it should be fabio:hasPublicationYear indeed!
